<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>975</width>
    <height>698</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl" native="true">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>20</y>
      <width>581</width>
      <height>471</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
      <horstretch>1</horstretch>
      <verstretch>1</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="Timeline" name="timeline_listWidget">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>20</y>
      <width>141</width>
      <height>301</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_8">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>630</y>
      <width>115</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Raytrace!</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>520</y>
      <width>581</width>
      <height>141</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="movable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Deformation</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>0</y>
        <width>61</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Axis</string>
      </property>
      <widget class="QRadioButton" name="radioButton_axis_z">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_axis_x">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_axis_y">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>0</y>
        <width>141</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Deformation Type</string>
      </property>
      <widget class="QRadioButton" name="radioButton_twist">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>131</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Twist</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_squash">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>121</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Squash/Stretch</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_bend">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Bend</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_taper">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Taper</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>131</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Control point position</string>
      </property>
      <widget class="QDoubleSpinBox" name="latticeZ_spinbox">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>80</y>
         <width>66</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="latticeX_spinbox">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>20</y>
         <width>66</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="latticeY_spinbox">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>50</y>
         <width>66</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_ctrl_x">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_ctrl_y">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_ctrl_z">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>0</y>
        <width>221</width>
        <height>41</height>
       </rect>
      </property>
      <property name="title">
       <string>Deformation Amount</string>
      </property>
      <widget class="QSlider" name="deform_slider">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>201</width>
         <height>22</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Tab 2</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>221</width>
      <height>671</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>Mesh</string>
     </attribute>
     <widget class="QCheckBox" name="checkBox">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>490</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Sharpen</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>HalfEdges</string>
      </property>
     </widget>
     <widget class="QListWidget" name="vertex_listWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>59</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertices</string>
      </property>
     </widget>
     <widget class="QListWidget" name="edge_listWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
     </widget>
     <widget class="QListWidget" name="face_listWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>300</y>
        <width>191</width>
        <height>121</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>280</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Faces</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>430</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Import .obj</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_4">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>430</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Subdivide</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>460</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Vertex</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_3">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>460</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete Vertex</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>490</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Triangulate Quad</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>550</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>530</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Face color</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>580</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>610</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>B</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>580</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>580</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>530</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertex position</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>580</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>G</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>550</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>610</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>610</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>610</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>550</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
      <property name="accessibleName">
       <string/>
      </property>
      <property name="suffix">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>550</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>R</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_6">
     <attribute name="title">
      <string>Skeleton</string>
     </attribute>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Joints</string>
      </property>
     </widget>
     <widget class="QTreeWidget" name="treeWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>30</y>
        <width>201</width>
        <height>221</height>
       </rect>
      </property>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
     <widget class="QPushButton" name="pushButton_6">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>250</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Import .json</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_7">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>280</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>skin</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>430</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_9">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>370</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>349</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Position</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>430</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>370</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_8">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>430</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>400</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>400</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_10">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>400</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>430</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_11">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>370</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>350</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Rotation</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_7">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>400</y>
        <width>66</width>
        <height>24</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_29">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>330</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QListWidget" name="meshList">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>370</y>
      <width>141</width>
      <height>192</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_21">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>350</y>
      <width>59</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Meshes</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>975</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QWidget</extends>
   <header>mygl.h</header>
   <slots>
    <signal>sig_SendRootNode(QTreeWidgetItem*)</signal>
    <signal>sig_SendRaycastSelection(QTreeWidgetItem*)</signal>
    <signal>sig_populateVert(std::vector*)</signal>
    <signal>sig_populateEdge(QListWidgetItem*)</signal>
    <signal>sig_populateFace(QListWidgetItem*)</signal>
    <signal>sig_populateVert(QListWidgetItem*)</signal>
    <signal>sig_clearList()</signal>
    <signal>sig_addTreeRoot(QTreeWidgetItem*)</signal>
    <signal>sig_selectLatticeVertex()</signal>
    <signal>sig_set_meshList()</signal>
    <slot>slot_ReceiveSelectedNode(QTreeWidgetItem*)</slot>
    <slot>slot_raytrace()</slot>
    <slot>slot_mesh_selected(QListWidgetItem*)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Timeline</class>
   <extends>QListWidget</extends>
   <header>timeline.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>mygl</sender>
   <signal>sig_populateVert(QListWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_populateVert(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_clearList()</signal>
   <receiver>edge_listWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_clearList()</signal>
   <receiver>vertex_listWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_clearList()</signal>
   <receiver>face_listWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>203</x>
     <y>468</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_addTreeRoot(QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_addTreeRoot(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>281</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>mygl</receiver>
   <slot>slot_raytrace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>964</x>
     <y>683</y>
    </hint>
    <hint type="destinationlabel">
     <x>690</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_selectLatticeVertex()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_populateLatticeSpinboxes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>783</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>883</x>
     <y>494</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mygl</sender>
   <signal>sig_set_meshList()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_set_meshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>284</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>524</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>meshList</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>mygl</receiver>
   <slot>slot_mesh_selected(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>858</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>705</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slot_populateVert(std::vector*)</slot>
  <slot>slot_populateVert(QListWidgetItem*)</slot>
  <slot>slot_addTreeRoot(QTreeWidgetItem*)</slot>
  <slot>slot_populateLatticeSpinboxes()</slot>
  <slot>slot_set_meshList()</slot>
 </slots>
</ui>
